In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt 

READ AIRPORTS DATA FROM CSV FILE

In [2]:
airport = pd.read_csv('airports.csv')

In [3]:
airport.head()

faa                           name        lat        lon   alt  tz dst  \
0  04G              Lansdowne Airport  41.130472 -80.619583  1044  -5   A   
1  06A  Moton Field Municipal Airport  32.460572 -85.680028   264  -6   A   
2  06C            Schaumburg Regional  41.989341 -88.101243   801  -6   A   
3  06N                Randall Airport  41.431912 -74.391561   523  -5   A   
4  09J          Jekyll Island Airport  31.074472 -81.427778    11  -5   A   

              tzone  
0  America/New_York  
1   America/Chicago  
2   America/Chicago  
3  America/New_York  
4  America/New_York

CHECK ALL UNIQUE VALUE FOR TZONE COLUMN

In [4]:
airport.tzone.unique()

array(['America/New_York', 'America/Chicago', 'America/Los_Angeles',
       'America/Vancouver', 'America/Phoenix', 'America/Anchorage',
       'America/Denver', 'Pacific/Honolulu', 'Asia/Chongqing', nan],
      dtype=object)

SELECT ONLY AMERICA TIME ZONE, EXCLUDE HONOLULU BECAUSE IT IS IN SOUTH WEST OF AMERICAN.

In [5]:
ap = airport[airport["tzone"].str.contains("America")==True]
ap.tzone.unique()

array(['America/New_York', 'America/Chicago', 'America/Los_Angeles',
       'America/Vancouver', 'America/Phoenix', 'America/Anchorage',
       'America/Denver'], dtype=object)

SORT BY LATITUDE TO FIND OUT WHICH AIRPORT HAS THE HIGHEST VALUE, WHICH REPRESENT THE NORTHERNMOST.

In [6]:
ap.sort_values(by=['lat'],ascending=False)


faa                                        name        lat         lon  \
230   BRW                  Wiley Post Will Rogers Mem  71.285446 -156.766003   
110   AIN                          Wainwright Airport  70.638056 -159.994722   
708   K03                               Wainwright As  70.613378 -159.860350   
152   ATK  Atqasuk Edward Burnell Sr Memorial Airport  70.467300 -157.436000   
1363  UUK                        Ugnu-Kuparuk Airport  70.330833 -149.597500   
...   ...                                         ...        ...         ...   
1002  OCA                                   Key Largo  25.325393  -80.274775   
926   MTH               Florida Keys Marathon Airport  24.726111  -81.051389   
1133  RBN                              Fort Jefferson  24.616670  -82.866670   
978   NQX                                Key West Nas  24.575834  -81.688889   
462   EYW                               Key West Intl  24.556111  -81.759556   

      alt  tz dst              tzone  
230    44  -9   A  America/Anchorage  
110    41  -9   A  America/Anchorage  
708    35  -9   A  America/Anchorage  
152    96  -9   A  America/Anchorage  
1363   67  -9   A  America/Anchorage  
...   ...  ..  ..                ...  
1002    8  -5   A   America/New_York  
926     7  -5   A   America/New_York  
1133    0  -5   A   America/New_York  
978     6  -5   A   America/New_York  
462     3  -5   A   America/New_York  

[1435 rows x 8 columns]

https://goo.gl/maps/6aSuryDpmbiU9wEF6
BY CHECKING ON THE GOOGLE MAP, THE WILEY POST WILL ROGGERS MEM IS LOCATED IN NORTHERN MOST OF ALASKA. 
__CONFIRM THE RESULT IS CORRECT.

_____________________________________________________
TO FIND OUT THE EASTERN MOST AIRPORT, WE NEED LOOK INTO THE AIRPORTS'S LONGTITUDE. 

In [7]:
ap.sort_values(by=['lon'],ascending=False)


faa                                 name        lat         lon  alt  \
1290  SYA                         Eareckson As  52.712275  174.113620   98   
444   EPM           Eastport Municipal Airport  44.910111  -67.012694   45   
624   HUL                         Houlton Intl  46.123083  -67.792056  489   
259   CAR                         Caribou Muni  46.871500  -68.017917  626   
1101  PQI  Northern Maine Rgnl At Presque Isle  46.688958  -68.044797  534   
...   ...                                  ...        ...         ...  ...   
1251  SNP                       St Paul Island  57.167333 -170.220444   63   
1281  SVA                     Savoonga Airport  63.686400 -170.493000   53   
522   GAM                      Gambell Airport  63.766800 -171.733000   27   
112   AKB                         Atka Airport  52.220278 -174.206389   56   
95    ADK                         Adak Airport  51.878000 -176.646000   18   

      tz dst              tzone  
1290  -9   A  America/Anchorage  
444   -5   A   America/New_York  
624   -5   A   America/New_York  
259   -5   A   America/New_York  
1101  -5   A   America/New_York  
...   ..  ..                ...  
1251  -9   A  America/Anchorage  
1281  -9   A  America/Anchorage  
522   -9   A  America/Anchorage  
112   -9   A  America/Anchorage  
95    -9   A  America/Anchorage  

[1435 rows x 8 columns]

EARECKSON AS IS ONLY RESULT THAT HAS POSTIVE VALUE, THIS MAKE ME TO THINK IF THERE'S ANY ISSUE WITH THIS DATA. AFTER CHECKING ON MAP, I FOUND THIS IS LOCATED IN FURTHER WEST OF THE UNITED STATE, AND IT IS IN BOUNDARY OF EASTERN WORLD. THAT'S WHY IT HAS POSTIVE VALUE. THUS WE NEED TO SKIP ALL POSTIVE RESULT.

https://goo.gl/maps/ixnm2Dai3NL6hm3aA
EASTPORT MUNICIPAL AIRPORT IS THE SECOND RESULT, IT HAS THE SMALLEST NEGATIVE VALUE, AFTER CHECKING ON THE MAP, IT IS LOCATED IN THE MAINE, EASTERN MOST TO THE BOUNDARY OF THE UNITED STATES. 
__CONFIRM EASTPORT MUNICIPAL AIRPORT IS THE CORRECT ANSWER.

_____________________________________________________
READ WEATHER FILE

In [8]:
weather = pd.read_csv('weather.csv')

In [9]:
weather.head()

origin  year  month  day  hour   temp   dewp  humid  wind_dir  wind_speed  \
0    EWR  2013      1    1     1  39.02  26.06  59.37     270.0    10.35702   
1    EWR  2013      1    1     2  39.02  26.96  61.63     250.0     8.05546   
2    EWR  2013      1    1     3  39.02  28.04  64.43     240.0    11.50780   
3    EWR  2013      1    1     4  39.92  28.04  62.21     250.0    12.65858   
4    EWR  2013      1    1     5  39.02  28.04  64.43     260.0    12.65858   

   wind_gust  precip  pressure  visib             time_hour  
0        NaN     0.0    1012.0   10.0  2013-01-01T06:00:00Z  
1        NaN     0.0    1012.3   10.0  2013-01-01T07:00:00Z  
2        NaN     0.0    1012.5   10.0  2013-01-01T08:00:00Z  
3        NaN     0.0    1012.2   10.0  2013-01-01T09:00:00Z  
4        NaN     0.0    1011.9   10.0  2013-01-01T10:00:00Z

In [10]:
weather.origin.unique()

array(['EWR', 'JFK', 'LGA'], dtype=object)

In [11]:
weather.dtypes

origin         object
year            int64
month           int64
day             int64
hour            int64
temp          float64
dewp          float64
humid         float64
wind_dir      float64
wind_speed    float64
wind_gust     float64
precip        float64
pressure      float64
visib         float64
time_hour      object
dtype: object

FILTER ALL FEB 12 2013 FLIGHTS.

In [12]:
wt = weather[(weather['year'] ==2013)& (weather['month']==2)& (weather['day'] == 12)]
wt.head()

origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
1006    EWR  2013      2   12     0  39.92  39.02  96.55     240.0   
1007    EWR  2013      2   12     1  39.92  37.94  92.56     250.0   
1008    EWR  2013      2   12     2  39.92  28.04  62.21     270.0   
1009    EWR  2013      2   12     3  39.02  26.96  61.63     260.0   
1010    EWR  2013      2   12     4  39.02  26.96  64.29     280.0   

      wind_speed  wind_gust  precip  pressure  visib             time_hour  
1006     6.90468        NaN     0.0    1006.9   10.0  2013-02-12T05:00:00Z  
1007     9.20624        NaN     0.0    1007.2   10.0  2013-02-12T06:00:00Z  
1008    20.71404   25.31716     0.0    1007.8   10.0  2013-02-12T07:00:00Z  
1009  1048.36058        NaN     0.0    1008.3   10.0  2013-02-12T08:00:00Z  
1010    12.65858        NaN     0.0    1008.8   10.0  2013-02-12T09:00:00Z

In [13]:
wt.sort_values(by=['wind_speed'],ascending=False)

origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
1009     EWR  2013      2   12     3  39.02  26.96  61.63     260.0   
18417    LGA  2013      2   12     2  42.98  26.06  50.94     290.0   
1018     EWR  2013      2   12    12  44.06  26.06  48.87     270.0   
18428    LGA  2013      2   12    13  44.06  23.00  43.02     300.0   
18429    LGA  2013      2   12    14  44.06  23.00  43.02     300.0   
...      ...   ...    ...  ...   ...    ...    ...    ...       ...   
1006     EWR  2013      2   12     0  39.92  39.02  96.55     240.0   
1027     EWR  2013      2   12    21  35.96  24.08  61.67     300.0   
18415    LGA  2013      2   12     0  42.08  37.40  86.89       NaN   
9709     JFK  2013      2   12     0  42.08  37.04  82.17     280.0   
1029     EWR  2013      2   12    23  35.96  23.00  58.94     250.0   

       wind_speed  wind_gust  precip  pressure  visib             time_hour  
1009   1048.36058        NaN     0.0    1008.3   10.0  2013-02-12T08:00:00Z  
18417    23.01560   31.07106     0.0    1007.1   10.0  2013-02-12T07:00:00Z  
1018     21.86482   31.07106     0.0    1012.5   10.0  2013-02-12T17:00:00Z  
18428    21.86482   25.31716     0.0    1011.7   10.0  2013-02-12T18:00:00Z  
18429    20.71404   25.31716     0.0    1011.5   10.0  2013-02-12T19:00:00Z  
...           ...        ...     ...       ...    ...                   ...  
1006      6.90468        NaN     0.0    1006.9   10.0  2013-02-12T05:00:00Z  
1027      5.75390        NaN     0.0    1013.0   10.0  2013-02-13T02:00:00Z  
18415     4.60312        NaN     0.0       NaN   10.0  2013-02-12T05:00:00Z  
9709      4.60312        NaN     0.0    1007.1   10.0  2013-02-12T05:00:00Z  
1029      4.60312        NaN     0.0    1013.7   10.0  2013-02-13T04:00:00Z  

[72 rows x 15 columns]

THE RESULT SHOWS EWR HAS THE WINDIEST WEATHER, WHICH THE WIND SPEED IS 1048.36. BUT AFTER VERIFIED ONLINE, THIS SPEED IS WAY OVER CURRENT WINDIEST SPEED RECORD. THUS THERE MUST BE SOME ERROR ON THIS NUMBER. 

__SO LGA WITH 23.01560 WAS THE WINDIEST WEATHER ON THAT DAY. 

1. What is the northernmost airport in the United States?   -- Wiley Post Will Rogers Mem
2. What is the easternmost airport in the United States?  -- Eastport Municipal Airport	
3. On February 12th, 2013, which New York area airport had the windiest weather?  -- LGA